docs/layout/dynamic-font-scaling.md

```diff
@@ -0,0 +1,205 @@
+
+```html
+<div class="parent">
+  Parent element with 20px
+  <div class="child">
+    Child element with 40px
+    <div class="child">Child element with 80px</div>
+  </div>
+</div>
+```
+
+<div style={{ fontSize: '20px' }}>
+  Parent element with 20px
+  <div style={{ fontSize: '2em' }}>
+    Child element with 40px
+    <div style={{ fontSize: '2em' }}>Child element with 80px</div>
+  </div>
+</div>
+
+Due to this compounding effect, we strongly recommend using `rem` units instead of `em` units when working with Dynamic Font Scaling. `rem` units set the font size of an element relative to the font size of the root element, which is typically `<html>`. The default font size of the root element is typically `16px`.
+
+In the following example, the computed font size of `.child` is `32px` because the font size is being computed relative to `html`, not `.parent`.
+
+```css
+.parent {
+  font-size: 20px;
+}
+
+.child {
+  font-size: 2rem;
+}
+```
+
+## How Dynamic Font Scaling works in Ionic
+
+All Ionic components that define font sizes use [rem units](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#lengths). This sizes the text in each component relative to the font size of the root element, which is usually the `html` element. This means that as the root element's font size changes, the text in all Ionic components scale in a consistent manner. This avoids the need to manually override each component's font size.
+
+### iOS
+
+Dynamic Font Scaling in Ionic builds on top of an iOS feature called [Dynamic Type](https://developer.apple.com/documentation/uikit/uifont/scaling_fonts_automatically#overview). To do this, Ionic sets the [font](https://developer.mozilla.org/en-US/docs/Web/CSS/font) of the root element to an Apple-defined text style. For consistency, Ionic uses the [body](https://developer.apple.com/documentation/uikit/uifont/textstyle/1616682-body) text style.
+
+Using the Apple-defined text style enables Dynamic Type, allowing all text in Ionic components to scale according to the system-level preference. Note that these Apple-defined fonts only work on Apple devices. As a result, these fonts will not work on Android devices even if your app is using `'ios'` mode.
+
+Ionic follows [Apple's Human Interface Guidelines for Typography](https://developer.apple.com/design/human-interface-guidelines/typography) when an app is in `'ios'` mode. As a result, important content is prioritized when the text size changes. This means a few things:
+
+1. Content in an `ion-header` or an `ion-footer` will have maximum font sizes to prioritize text in `ion-content` which is deemed more important than text in `ion-header` and `ion-footer`.
+2. Components such as `ion-badge` and `ion-back-button` will have minimum font sizes so they remain readable.
+3. Text in components such as `ion-tab-bar` and `ion-picker` do not participate in Dynamic Font Scaling according to Apple's Human Interface Guidelines.
+
+### Android
```

Comment:
I'm confused by this entire section. We aren't using `min`, `max` or `clamp` on Android values in Ionic and this section is under `How Dynamic Font Scaling works in Ionic`. Would this be better suited as info under `Using Dynamic Font Scaling` or a section with info on how the different devices / operating systems handle dynamic scaling?

Reply:
Our APIs are built on top of the browser behavior, so how Ionic behaves is influenced by the browser behavior which is why I had this section here. I mentioned min/max/clamp because developers potentially will need to use additional CSS to ensure their custom components support Dynamic Font Scaling. I wanted to call it out because developers may think the min/max/clamp behavior is an Ionic bug.